docs/madr/decisions/046-meshservice-policy-matching.md

```diff
@@ -0,0 +1,191 @@
+# Policy matching with MeshService
+
+* Status: accepted
+
+Technical Story: #9707
+
+## Context and Problem Statement
+
+Now that MeshService exists, we need to enable users to apply policies in a way that works
+with the new `MeshService` resource. This includes using `StatefulSet`.
+Additionally, policy should be appliable to `MeshMultiZoneService`.
+The inspect API should also reflect policies that use `MeshService`
+which will be covered in a separate MADR.
+
+Some things to remember:
+
+* Currently `kind: MeshService` in `from[].targetRef` is deprecated.
+* While `kind: MeshService` in `spec.targetRef` points to both a Dataplane and a set of inbounds,
+  it is not a canonical reference.
+  Its effects effectively mix with any policies applied
+  to any `MeshService` that points to this `Dataplane`/inbound.
+  A canonical reference can only be a direct choice of an entire `Dataplane` object
+  and potentially a subset of inbounds.
+* A `MeshService` only has a 1-1 relationship when used as `to[].targetRef.kind`
+  since then it corresponds to a particular Envoy outbound listener/some number of clusters.
+
+## Decision Drivers
+
+* Targeting headless `MeshService`
+* Targeting `MeshMultiZoneService`
+* A balance between correctness and ease of use when configuring policy.
+
+## Considered Options
+
+This MADR adds 2 kinds `to[].targetRef.kind` and Mesh*Route `backendRefs[].kind`:
+
+* `MeshService`
+* `MeshMultiZoneService`
+
```

Comment:
Hi Mike, could you please explain more with this sentence?

In the past, I learned from the [Target-Resources](https://kuma.io/docs/2.7.x/policies/targetref/#target-resources)
The Sidecar targetRef `to[].targetRef.kind` already allows kind `MeshService`, we repeat it again here?
And what's meaning of  "and Mesh*Route `backendRefs[].kind`"?  

Sorry that I did't get it.

Reply:
Yeah it's badly phrased, it adds new functionality for matching real `MeshServices` to the existing `kind: MeshService`.

> And what's meaning of "and Mesh*Route backendRefs[].kind"?

That refers to references in MeshHTTPRoute/MeshTCPRoute